pkg/storage/azure/azureclient/azureclient.go

```diff
@@ -81,63 +81,78 @@ func New(opts *Options) (*Client, error) {
 	}
 	coreOpts.PerCallPolicies = opts.Policies
 	creds := opts.Creds
-	if creds == nil {
-		var err error
-
-		// Managed Identity Override for ARO HCP
-		managedIdentityClientID := os.Getenv("ARO_HCP_MI_CLIENT_ID")
-		if managedIdentityClientID != "" {
-			options := azidentity.ManagedIdentityCredentialOptions{
-				ClientOptions: azcore.ClientOptions{
-					Cloud: cloudConfig,
-				},
-				ID: azidentity.ClientID(managedIdentityClientID),
-			}
-			creds, err = azidentity.NewManagedIdentityCredential(&options)
-			if err != nil {
-				return nil, err
-			}
-		} else if strings.TrimSpace(opts.ClientSecret) == "" {
-			options := azidentity.WorkloadIdentityCredentialOptions{
-				ClientOptions: coreOpts,
-				ClientID:      opts.ClientID,
-				TenantID:      opts.TenantID,
-				TokenFilePath: opts.FederatedTokenFile,
-			}
-			creds, err = azidentity.NewWorkloadIdentityCredential(&options)
-			if err != nil {
-				return nil, err
-			}
-		} else {
-			options := azidentity.ClientSecretCredentialOptions{
-				ClientOptions: coreOpts,
-			}
-			creds, err = azidentity.NewClientSecretCredential(
-				opts.TenantID,
-				opts.ClientID,
-				opts.ClientSecret,
-				&options,
-			)
-			if err != nil {
-				return nil, err
-			}
-		}
-	}
-
 	coreOpts.Retry = policy.RetryOptions{
 		MaxRetries: -1, // try once
 	}
 
 	return &Client{
-		creds:          creds,
-		clientOpts:     &coreOpts,
-		tagset:         opts.TagSet,
-		subscriptionID: opts.SubscriptionID,
+		creds:      creds,
+		clientOpts: &coreOpts,
+		opts:       opts,
 	}, nil
 }
 
+func (c *Client) getCreds() (azcore.TokenCredential, error) {
```

Comment:
`getCreds()` will also be needed when `cfg.key` is not present in `NewBlobClient`

Reply:
great catch, fixed!